core/src/main/java/org/apache/accumulo/core/fate/ZooStore.java

```diff
@@ -130,6 +136,31 @@ public long create() {
     }
   }
 
+  /*
+   * Holds a copy of all of the fate transaction in zookeeper. Used for finding the next one to
+   * reserve. When empty a single thread will refill. All fate threads can pull off of the queue as
+   * they look for something to reserve. This is used so that each thread does not have to read all
+   * children from zookeeper when looking for any fate tx to reserve.
+   */
+  private final Queue<String> reservationCandidates = new ConcurrentLinkedQueue<>();
+  private final Lock reservationCandidateLock = new ReentrantLock();
```

Comment:
Could we just sync on the zoostore/this instead? Would be nice to avoid a second lock